akka-remote/src/test/scala/akka/remote/artery/tcp/ssl/RotatingKeysSSLEngineProviderSpec.scala

```diff
@@ -269,7 +267,10 @@ abstract class RotatingKeysSSLEngineProviderSpec(extraConfig: String)
   override def beforeTermination(): Unit = {
     systemsToTerminate.foreach { systemToTerminate =>
       system.log.info(s"Terminating $systemToTerminate...")
-      systemToTerminate.terminate()
+      val now = System.nanoTime()
+      TestKit.shutdownActorSystem(systemToTerminate, verifySystemShutdown = true)
+      val lasted = System.nanoTime() - now
+      system.log.info(s"Terminated $systemToTerminate after ${lasted / 1000000} ms")
     }
     systemsToTerminate.foreach(shutdown(_, verifySystemShutdown = true))
```

Comment:
Now it looks like it does the same twice?
I think the intention with the first loop was to terminate all at the same time asynchronously and then wait for completion here. Not sure if that really solved anything.

Reply:
You're right of course - I'll keep this intact for now (though indeed I'm not sure if it'd help, and it makes the logs harder to follow - so if we confirm it doesn't solve things it might be good to terminate them sequentially again)